File: 2026_GFG_POTD/JANUARY/01-01-2026/Intersection in Y Shaped Lists.py
'''
class Node:
    def __init__(self, data):
        self.data = data
        self.next = None

'''

class Solution:
    def intersectPoint(self, head1, head2):
        a, b = head1, head2
        while a is not b:
            a = a.next if a else head2
            b = b.next if b else head1
        return a

File: 2026_GFG_POTD/JANUARY/01-01-2026/Intersection in Y Shaped Lists.cs
/*
class Node {
    public int data;
    public Node next;

    public Node(int x) {
        data = x;
        next = null;
    }
}
*/

class Solution {
    public Node intersectPoint(Node head1, Node head2) {
        Node a = head1, b = head2;
        while (a != b) {
            a = (a != null) ? a.next : head2;
            b = (b != null) ? b.next : head1;
        }
        return a;
    }
}